src/components/pageLayout/PageLayout.tsx

```diff
@@ -1,18 +1,29 @@
 import Footer from "@/components/footer/Footer";
 import Header from "@/components/header/Header";
+import { auth, signIn } from "@/auth/auth";
+import Dialog from "../dialog/Dialog";
 
-export default function PageLayout({
+export default async function PageLayout({
   children,
 }: Readonly<{
   children: React.ReactNode;
 }>) {
+  const session = await auth();
+  const user = session?.user;
+
+  const handleLogin = async () => {
+    "use server";
+    await signIn();
+  };
+
   return (
     <div className="page-layout-container">
-      <Header />
+      <Header handleLogin={handleLogin} isLoggedIn={!!user} />
       <div className="max-w-screen-xl flex flex-wrap items-center justify-between mx-auto p-4">
         {children}
       </div>
       <Footer />
+      <Dialog loggedIn={user === undefined ? false : true} />
```

Comment:
We could simplify this one using the Boolean object - so if the object exists, it should return true, otherwise undefined or null return false

```suggestion
      <Dialog loggedIn={Boolean(user)} />
```